pkg/task/ecec.go

```diff
@@ -0,0 +1,92 @@
+/*
```

Comment:
I would like to deviate from existing package structure.
Will it be possible to make use of idiomatic maya packaging? e.g. below comes to my mind
- verify the package path
- verify the naming
- verify the patterns

```go
// in pkg/kubernetes/podexec/v1alpha1/podexec.go

import (
  "github.com/pkg/errors"
)

type podexec struct {
  object             *api_core_v1.PodExecOptions
  ignoreErrors   bool
  errs                []err
}

func (p *podexec) AsAPIPodExec() (*api_core_v1.PodExecOptions, error) {
  err = p.Validate()
  if err != nil && !p.ignoreErrors {
    return nil, err
  }
  return p.object, nil
}

func WithTemplate(context, yaml string, values map[string]interface{}) (*podexec, error) {
  b, err := template.AsTemplatedBytes(context, yaml, values)
  if err != nil {
    return nil, err
  }
  var exec api_core_v1.PodExecOptions
  err = yaml.Unmarshal(b, &exec)
  if err != nil {
    return nil, err
  }
  p.object = exec
  return p, nil
}

func (p *podexec) Validate() error {
  if len(p.errs) != 0 {
    return errors.Errorf("validation failed: %v", p.errs)
  }
  if p.object.Command == "" {
    return errors.New("validation failed: command not provided")
  }
  if p.object.ContainerName == "" {
    return errors.New("validation failed: container name not provided")
  }
  return nil
}

type buildOption func(*podexec)

func IgnoreErrors() buildOption {
  return func(p *podexec) {
    p.ignoreErrors = true
  }
}

func (p *podexec) Apply(opts ...buildOption) (*podexec, error) {
  for _, o := range opts {
    o(p)
  }
  return p
}
```

Reply:
done please have a look